javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/OpenTelemetryInstaller.java

```diff
@@ -46,8 +46,6 @@ public static synchronized void installAgentTracer(Config config) {
       if (config.getBooleanProperty(JAVAAGENT_NOOP_CONFIG, false)) {
         GlobalOpenTelemetry.set(NoopOpenTelemetry.getInstance());
       } else {
-        System.setProperty("io.opentelemetry.context.contextStorageProvider", "default");
```

Comment:
After taking a step back, I'm sort of wondering how bad this approach was actually.

- We force the context storage in the app classloader to be the bridge into the agent's context storage
- Any attempt to set this by the user will not work as expected since it's only used in the agent classloader

So just hardcoding this property doesn't seem that bad anymore as regardless of whether we have the noop config or not, it seems like it could have been a good idea. Any thoughts?

Reply:
I agree that since even the name of system property is renamed by shading it is unlikely to clash with whatever user is doing.
I don't like `System.setProperty` approach much because seeing it immediately makes me wonder why would you do something like that. In that sense making `NoopContextStorageProvider` non discoverable if we don't want it to be found feels like a more natural solution, but considering that `System.setProperty` only needs 1 line it doesn't feel that bad any more.

Reply:
I agree with the hesitance around setProperty for control flow. But indeed lacking any one other identifiable problem, the 1 line aspect is very appealing here.

Reply:
👍

Reply:
@anuraaga @trask @laurit I am confused about what is our final decision :)

Reply:
can excluding the META-INF/services file be one line also, by adding the exclusion directly when building the javaagent, e.g. the `==>` line below

```
fun isolateSpec(projectsWithShadowJar: Collection<Project>): CopySpec = copySpec {
  from(projectsWithShadowJar.map { zipTree(it.tasks.getByName<ShadowJar>("shadowJar").archiveFile) }) {
    // important to keep prefix "inst" short, as it is prefixed to lots of strings in runtime mem
    into("inst")
    rename("""(^.*)\.class$""", "$1.classdata")
    // Rename LICENSE file since it clashes with license dir on non-case sensitive FSs (i.e. Mac)
    rename("""^LICENSE$""", "LICENSE.renamed")
==> exclude("META-INF/services/io.opentelemetry.javaagent.shaded.io.opentelemetry.context.ContextStorageProvider")
  }
}
```

Reply:
@trask this is what I did initially. You actually need to do this for both testing agent and real agent. As this file also gets shaded into intermediate dependencies like `opentelemetry-instrumentation-1.5.0-SNAPSHOT-all.jar` it gets hard to track where it comes from. This is why I chose to eliminate it as close to the source as possible.